apps/frontend/src/pages/CreateAccount.tsx

```diff
@@ -1,5 +1,206 @@
+import React, { useState } from "react";
+import {
+  Box,
+  Button,
+  TextField,
+  Typography,
+  FormHelperText,
+  Link,
+} from "@mui/material";
+import martletPlaceLogo from "../images/martletplace-logo.png";
+import { colors } from "../styles/colors";
+import { useStyles } from "../styles/pageStyles";
+import { useNavigate } from "react-router-dom";
+import axios from "axios";
+
 const CreateAccount = () => {
-  return <h1>CreateAccount</h1>;
+  const classes = useStyles();
+  const [email, setEmail] = useState("");
+  const [name, setName] = useState("");
+  const [username, setUsername] = useState("");
+  const [password, setPassword] = useState("");
+
+  const [emailError, setEmailError] = useState("");
+  const [usernameError, setUsernameError] = useState("");
+  const [passwordError, setPasswordError] = useState("");
+
+  const isFormEmpty = !email || !name || !username || !password;
+
+  const navigate = useNavigate();
+
+  const handleCreateAccount = async (e: React.FormEvent) => {
+    e.preventDefault();
+
+    // Regex for UVic email validation
+    const emailFormat = /^[^@]+@uvic\.ca$/;
+
+    // Regex for username validation
+    const usernameFormat = /^[a-zA-Z]{1,20}$/;
```

Comment:
Any reason to not include numbers also?
```suggestion
    const usernameFormat = /^[a-zA-Z0-9]{1,20}$/;
```

Reply:
Because of this comment from @Scott-Kenning :
https://github.com/UVicMartletplace/martletplace/pull/143#discussion_r1630688926